extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -384,6 +384,13 @@ public void configurationDescriptorBuilding(
             return;
         }
 
+        if (shouldIgnorePersistenceXmlResources(hibernateOrmConfig)) {
+            LOG.warnf("Hibernate ORM ignores 'persistence.xml' files because config property '%s' or System property '%s' was set to true",
+                    "quarkus.hibernate-orm.persistence-xml.ignore",
+                    "SKIP_PARSE_PERSISTENCE_XML");
+            return;
```

Comment:
Wait, the end of this method is about non-`persistence.xml` stuff, so returning early here will just break everything (and I mean *everything*).

Did this actually pass any test?

On top of that, this is not actually useful, because if we need to ignore `persistence.xml`, then `io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#parsePersistenceXmlDescriptors` will not produce any `PersistenceXmlDescriptorBuildItem` and the loop below will not do anything.

So really... I'd be inclined to investigate your problem a bit more.

Reply:
@yrodiere TBH, I hoped GHA CI will show me how this approach behaves.

I haven't had a chance to investigate it more, but when I spotted the evaluation of `PersistenceXmlDescriptors`, I thought it was unnecessary to go over them as we should only resolve the config properties provided in `application.properties`. 

But when I'm looking at it right now, this is not a good place to check whether these ignore props are set or not.

The thing is, that I'd assume that even when I have some `persistence.xml` files in my project and I explicitly specify to ignore them, they're really ignored and I'm able to process all build steps :)) Not happening right now.

Reply:
> The thing is, that I'd assume that even when I have some `persistence.xml` files in my project and I explicitly specify to ignore them, they're really ignored and I'm able to process all build steps :)) Not happening right now.

Yes, but the files should be ignored in `HibernateOrmProcessor#parsePersistenceXmlDescriptors`.

If they are not, there is probably a problem there. I need to debug your application and understand what's going on.

Reply:
Applications are not supposed to produce build items, so no.

I suppose you have a custom Quarkus extension and in that case yes, I suppose we could allow those to produce their own `PersistenceXmlDescriptor` through their own `PersistenceXmlDescriptorBuildItem` producers, though that's definitely not something I anticipated.

I'll skip the part where I rant about Quarkus' lack of clearly defined APIs and SPIs that are driving me insane. I suspect I'm never going to convince the team to change that.

Now, in your case... it seems you're using both `quarkus.hibernate-orm.*` properties and `persistence.xml` files provided through `PersistenceXmlDescriptorBuildItem`.

Two questions:

1. Could you shed some light on why you need custom `persistence.xml` files?
2. Are you aware most `quarkus.hibernate-orm.*` properties **will not affect your persistence units bound to `persistence.xml` files in any way**, and that it's something that we're not going to change in Quarkus? Is that not a problem?

Reply:
> I suppose you have a custom Quarkus extension and in that case yes, I suppose we could allow those to produce their own PersistenceXmlDescriptor through their own PersistenceXmlDescriptorBuildItem producers, though that's definitely not something I anticipated.

Yep, we have our own Quarkus extension :))

>Could you shed some light on why you need custom persistence.xml files?

I'll check with our Store team what's the exact purpose of having both of them present there. 

>Are you aware most quarkus.hibernate-orm.* properties will not affect your persistence units bound to persistence.xml files in any way, and that it's something that we're not going to change in Quarkus? Is that not a problem?

Yep, as I've said, I need to discuss the relevance of the `persistence.xml` files in our app.

@yrodiere Thanks for your help!

Reply:
No problem, thanks for digging. Please let us know (preferably on #31766 rather than on this PR) when you have more information and we'll see how we can proceed.

In the meantime I'll close this PR, as the solution is elsewhere :)